### ISSUE TITLE
Error: Table not created yet.

### ISSUE THREAD

**Issue by SuvabBaral** [animir/node-rate-limiter-flexible#78]
So, I have created a RateLimiterPostGres in the following way and tried to use it. But I am getting error when I use **consume.** However, if I check the database, a table is already created with **key, points and expire** columns.

![Screenshot from 2020-05-26 13-53-50](https://user-images.githubusercontent.com/54930322/82876997-58445380-9f59-11ea-9132-ecc7116e2da1.png)

Used the above as:

![Screenshot from 2020-05-26 13-54-11](https://user-images.githubusercontent.com/54930322/82877145-8de93c80-9f59-11ea-8ea9-15694738dc8b.png)

But getting the the table not created error:

![Screenshot from 2020-05-26 13-54-49](https://user-images.githubusercontent.com/54930322/82877257-a9544780-9f59-11ea-88ba-dee8885a8137.png)
Can you please help me out where I might be wrong and what can I do?

**Comment by animir** [comment]
@SuvabBaral Hi, rate limiter should check if table created or not, before you can use consume. that `ready` callback is called, when table exists query made. until then your rate limiter "thinks", table is not created yet. that's why you see that error.

I changed related example [here](https://github.com/animir/node-rate-limiter-flexible/wiki/PostgreSQL)

I think, you should export async function in your case. It should return a Promise resolved with rate limiter instance from ready callback.

**Comment by SuvabBaral** [comment]
@animir Yeah thanks for this. 
But i'm sorry I still couldn't grasp what you mean exactly in your last statement. My userId or ip exists in another file as you can see above. And from your refined example, I need to pass it as on the `ready` callback? I couldn't get what you mean by **return a Promise resolved with rate limiter instance from ready callback?**

**Comment by animir** [comment]
@SuvabBaral I understand, you want it in separate file. So you should not return rate limiter instance, but a function. 
Something like:

```javascript
module.exports.getMyRateLimiter = async () => {
  return new Promise((resolve, reject) => {
    let rateLimiter
    const ready = (err) => {
       if (err) {
         // handle error
       } else {
          resolve(rateLimiter)
       }
    }

    rateLimiter = new RateLimiterPostgres(opts, ready)
  })
}
```

I hope, this helps.

**Comment by animir** [comment]
@SuvabBaral Or you could add a new option `tableCreated` to `RateLimiterPostgres` class. If it is set to `true`, limiter doesn't make any checks. In this case `ready` callback won't be necessary.

**Comment by SuvabBaral** [comment]
Thanks @animir  returning a resolved rateLimiter instance works great!.
One question that I had in mind was, if we are using a serverless application,  it may happen that there is no running instance which will wait for 30 seconds (suppose the block duration is 30 seconds) to reset this points consumed(counter). 
What will happen in that case? Will it be reset next time the login is called?

**Comment by animir** [comment]
@SuvabBaral Data expiring happens on the store level. Does this answer your question? If not, could you describe it thoroughly.

**Comment by SuvabBaral** [comment]
Yes that actually answers it.
So, I tried to simulate this locally by consecutively increasing the count to 3 and closing the server.  Then after the time mentioned in duration elapsed,  I ran the server back on and tried a fail login attempt - which sets the counter at 1 again. Meaning it was reset and again increased to 1.
So the reset will happen in the next login, even if there is no running instance. 
This is great work! Thanks!
